db/python/tables/bq/billing_ar_batch.py

```diff
@@ -46,15 +46,24 @@ async def get_batches_by_ar_guid(
         # return empty list if no record found
         return None, None, []
 
-    async def get_ar_guid_by_batch_id(self, batch_id: str) -> str:
+    async def get_ar_guid_by_batch_id(
+        self, batch_id: str | None
+    ) -> tuple[datetime | None, datetime | None, str | None]:
         """
-        Get ar_guid for given batch_id
+        Get ar_guid for given batch_id,
+        if batch_id is found, but not ar_guid, then return batch_id back
         """
+        if batch_id is None:
+            return None, None, None
+
         _query = f"""
-        SELECT ar_guid
+        SELECT ar_guid,
+            MIN(min_day) as start_day,
+            MAX(max_day) as end_day
         FROM `{self.table_name}`
         WHERE batch_id = @batch_id
-        AND ar_guid IS NOT NULL
+        GROUP BY ar_guid
+        ORDER BY ar_guid DESC
         LIMIT 1;
```

Comment:
Why do an order by then LIMIT, maybe catch multiple, and then handle 0 (None), 1 (Single), many (Exception) to give a contextual exception for debugging?

Reply:
Good point. I have changed the code accordingly.
There are some jobs within batches, where is missing ar_guid, so when querying the materlized view we can get 2 records per one batch, one is valid ar-guid and the other is NULL as ar-guid. The ORDER BY and LIMIT will ignore that NULL and only choose the valid ar-guid if present.
I had removed the limit and check if the second record is not None, which would be a problem.
If second records is None, then this is fine as we have batches where some its job has no ar-guid set.

There are 8 batches like that in db so far:
SELECT batch_id, ar_guid
FROM `billing-admin-290403.billing_aggregate.aggregate_job_details` 
where batch_id in ('432372', '431078', '431923', '431174', '431590', '430639', '432673', '430868')

For instance, job 60 has no ar-guid
https://batch.hail.populationgenomics.org.au/batches/432673/jobs/60

However the rest of jobs of the same batch have it set, e.g. job#1
https://batch.hail.populationgenomics.org.au/batches/432673/jobs/1

It is not necessary an issue/bug we want to report, we can safely ignore it.

If there is more than one valid ar-guid then we have a problem and the new change would throw Exception. 
Atm we do not have records like that in database.